service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/AssignManagedIdentityAzureVmStep.java

```diff
@@ -37,17 +40,23 @@ public AssignManagedIdentityAzureVmStep(
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    // Note we only assign the VM to a managed identity if the resource has an assigned user.
-    // This step is a no-op otherwise.
-    if (resource.getAssignedUser().isPresent()) {
-      final AzureCloudContext azureCloudContext =
-          context
-              .getWorkingMap()
-              .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+
+    ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
+    MsiManager msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
 
-      ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
-      MsiManager msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
+    Set<String> userAssignedIdentities = new HashSet<>();
 
+    // Add user assigned identities from the request, if any
+    if (CollectionUtils.isNotEmpty(resource.getUserAssignedIdentities())) {
```

Comment:
It seems like we're consuming the list of identities and adding them to the VM without first checking with Sam, is that something that should be done? That is, can a user add any arbitrary user assigned identity to the vm using this api?

Reply:
I think the key factor is that the identities are only accepted for application-managed resources (WSM throws a Bad Request otherwise). The calling application (Leo) is requesting these identities from Sam as the user. WSM trusts the application to the validation.

Application-managed resources are tricky, because WSM does not have access to the user's token -- so we were thinking to do the Sam validation upstream of WSM for the application case.

Reply:
Added a comment to this effect as well